sdks/python/apache_beam/transforms/core.py

```diff
@@ -220,15 +222,29 @@ def __init__(self, generator, *args, **kwargs):
     self._gen_args = args
     self._gen_kwargs = kwargs
 
+  def element_generator(self):
+    while self._current_value is not None:
+      yield self._current_value
+
+    raise StopIteration()
+
   def start_bundle(self, context):
-    self._generator = self._generator_def(*self._gen_args, **self._gen_kwargs)
-    self._generator.send(None)
+    self._generator = self._generator_def(
+        self.element_generator(), *self._gen_args, **self._gen_kwargs)
 
   def process(self, context, *args, **kwargs):
-    return self._generator.send((context, args, kwargs))
+    self._current_value = (context, args, kwargs)
```

Comment:
Fixed in https://github.com/elibixby/incubator-beam/pull/2/commits/c1a0fbecac1d76cb5b8e88d52bb7485ef45694d1 with a small exception:

In python there's no obvious was to me to emit elements in a `finish_bundle` call. IIRC there was a way in the Java SDK. This would be nice for safety but isn't strictly necessary